frontend/src/components/JoinContainerSeparator/JoinContainerSeparator.spec.tsx

```diff
@@ -0,0 +1,17 @@
+import { render, screen } from '@testing-library/react';
+import { describe, it, expect } from 'vitest';
+import JoinContainerSeparator from './JoinContainerSeparator';
+
+describe('JoinContainerSeparator', () => {
+  it('renders the component with separators and text and applies correct styling', () => {
+    render(<JoinContainerSeparator />);
+
+    expect(screen.getByText('or')).toBeInTheDocument();
+
+    const separators = screen.getAllByTestId('separator');
+    expect(separators).toHaveLength(2);
+
+    // checking here that the parent of a Separator component has the correct classNames
+    expect(separators[0].parentElement).toHaveClass('flex w-full items-center');
```

Comment:
I'm not the biggest fan of testing `className`s unless absolutely necessary, ie checking if the `hidden` class is present in some obscure test cases.

It seems brittle given we're testing the implementation (of tailwind) and not how the (parent) functions. I'm okay nixing these `toHaveClass` tests, or we could replace them with some styling checks. We could use `window.getComputedStyle` and check if we're using `display: flex`, `width: 100%`, `justify-items: center`, etc. instead.

We've also talked about moving away from tailwind in the past, so it's likely these `toHaveClass` tests become tech debt that we'll need to remove or replace.

Reply:
I think I was just getting creative with the testing here to improve the SonarCloud coverage. However, I feel like in context of testing our UI this helps ensuring that some unintended changes to CSS don't go unnoticed and won't accidentally break something. 
Testing for particular classes seems like a simpler approach, too. 
Moving away from tailwind is something we want to do but who knows when and if that happens.
I'll be curious to hear @maikthomas 's thoughts on this. again, I have no issues removing these expect statements.

Reply:
I see, thanks Mike. Given that it's a bit controversial and a bit unstable, I'll remove these 🙏